lightrag/lightrag/core/generator.py

```diff
@@ -263,8 +286,8 @@ async def acall(
         api_kwargs = self._pre_call(prompt_kwargs, model_kwargs)
         completion = await self.model_client.acall(
             api_kwargs=api_kwargs, model_type=self.model_type
```

Comment:
im thinking we need to have a new function, `self.model_client_call` and `model_client_acall` to call and parse the completion this way, the pre_call and post_call does not need to be async for now

Reply:
im working on something also need to separate it:

```
    def _model_client_call(self, api_kwargs: Dict) -> Any:
        # call the model client
        try:
            # check the cache
            index_content = json.dumps(api_kwargs)  # all messages
            cached_completion = self._check_cache(index_content)
            if cached_completion is not None:
                return cached_completion
            completion = self.model_client.call(
                api_kwargs=api_kwargs, model_type=self.model_type
            )
            # prepare cache
            self._save_cache(index_content, completion)
            return completion
        except Exception as e:
            log.error(f"Error calling the model: {e}")
            raise e
```

You can use this minus the cache, here is how to use it in the call

```
 output: GeneratorOutputType = None
        # call the model client

        completion = None
        try:
            completion = self._model_client_call(api_kwargs=api_kwargs)
        except Exception as e:
            log.error(f"Error calling the model: {e}")
            output = GeneratorOutput(error=str(e))
        # process the completion
        if completion:
            try:
                output = self._post_call(completion)

            except Exception as e:
                log.error(f"Error processing the output: {e}")
                output = GeneratorOutput(raw_response=str(completion), error=str(e))
 ```